Source/WebCore/workers/WorkerThread.cpp

```diff
@@ -144,14 +144,14 @@ Ref<Thread> WorkerThread::createThread()
 {
     if (is<WorkerMainRunLoop>(runLoop())) {
         // This worker should run on the main thread.
-        RunLoop::protectedMain()->dispatch([protectedThis = Ref { *this }] {
-            protectedThis->workerOrWorkletThread();
+        RunLoop::protectedMain()->dispatch([this, protectedThis = Ref { *this }] {
+            workerOrWorkletThread();
         });
         ASSERT(isMainThread());
         return Thread::currentSingleton();
     }
 
-    return Thread::create(threadName(), [this] {
+    return Thread::create(threadName(), [this, protectedThis = Ref { *this }] {
```

Comment:
It seems this change causes a bunch of crashes. Should we annotate this somehow because we're not on the main thread? (At least I saw elsewhere that's a reason not to use a Ref pointer and instead add an annotation.)

Reply:
You should resolve this change for now and use a suppression macro to silence the issue. It is not safe to destroy the WorkerThread on another thread than the main thread. The `WorkerOrWorkletThread::workerOrWorkletThread()` function also already takes care of protecting `this` at the beginning for safety and calls `destroyWorkerGlobalScope(WTFMove(protectedThis))` at the end, which expects a single ref for `this`. There might be a way to refactor the code further to follow Safer CPP while keeping the code functional but it will not be easy.